docs/main.md

```diff
@@ -0,0 +1,291 @@
+  * **utils:** Useful classes that might be directly related to the project, like parsing certain content, databases, etc. _(optional)_
+
+
+## Test Scripts
+
+`Test Scripts` are the base of `Marvin`'s execution process. They are structured in a way that they imitate a manual test case, and the main part are the steps involved in it.
+
+### Test Script File
+
+`Test Scripts` have three main methods: `setup`, `run` and `tear_down`.
+
+Here is a basic template for a new `Test Script`:
+
+```python
+# test_cases/example_test.py
+from marvin import TestScript
+
+
+class TestScriptExample(TestScript):
+    """Test Script Example for documentation"""
+
+    NAME = "Test Script Example"
+
+    def setup(self, data):
+        pass
+
+    def run(self, data):
+        pass
+
+    def tear_down(self, _data):
+        pass
+```
+
+Inside this methods is where the steps are called. For example, in the `setup` method, you would call steps that open a website, create a connection to a database, or anything that would be required to set-up the test environment. On the `tear_down` method, you would close any connection to the database, logout, or perform any clean up tasks. And in the `run` method it is where you should have the main logic of your test and this method is the one that can be executed one or more times, depending on the data defined for that test.
+
+### Data Files
+
+As we said, data files are consumed by `Data Providers` and `Marvin` has support for YAML/JSON format out of the box. But if you need your own data provider, you can create one. Check [Data Providers](docs/data_providers.md) documentation.
+
+Here is a example:
+
+```yaml
+# test_cases/example_test.yaml
+setup:
+  username: admin-test
+  password: hunter2
+
+iterations:
+  - create_user:
+      name: John Iteration 1
+      account_type: viewer
+    should_succeed: True
+
+  - create_user:
+      name: Mary Iteration 2
+      account_type: reporter
+    should_succeed: True
+
+  - create_user:
+      name: George Iteration 3
+      account_type: admin
+    should_succeed: False
+
+tear_down:
+  logout: true
+```
+
+The data file format resembles the structure that `Test Script` have and they should also have three section: `setup`, `iterations` and `tear_down`.
+
+The `setup` and `tear_down` sections matches the same as the ones on `Test Script` and the `iterations` section is a list that for every item inside that list, it will execute the `run` method on the `Test Script`.
+
+### Test Scripts and Data Files relationship
+
+To show you how `Marvin` links the `Test Scripts` and the `Data Files`, here is a little example following the data above:
+
+```python
+# test_cases/example_test.py
+from marvin import TestScript
+
+
+class CreateUser(TestScript):
+    """Step for creating a user specifying a user type"""
+
+    NAME = "Create User"
+
+    def setup(self, data):
+        print "Using username {}".format(data["username"])
```

Comment:
RIght, also in the docs let's be python3 compliant. let's use the print function instead of the print statement. I.e `print(...)`